src/screens/WalletConnect/Provider.js

```diff
@@ -0,0 +1,280 @@
+        setCurrentCallRequestId(payload.id);
+        navigate(NavigatorName.SignMessage, {
+          screen: ScreenName.SignSummary,
+          params: {
+            message: wcCallRequest.data,
+            accountId: account.id,
+          },
+        });
+      } else {
+        connector.rejectRequest({
+          id: payload.id,
+          error: {
+            message: "Request not supported",
+          },
+        });
+      }
+    });
+
+    setConnector(connector);
+    if (connector.connected) {
+      setDappInfo(session.session.peerMeta);
+      setStatus(STATUS.CONNECTED);
+    }
+  };
+
+  const disconnect = () => {
+    if (connector) {
+      connector.killSession();
+    }
+
+    setDappInfo();
+    setApproveSession();
+    setSession({});
+    setError();
+    setStatus(STATUS.DISCONNECTED);
+    setConnector();
+  };
+
+  useEffect(() => {
+    if (!(currentCallRequestResult || currentCallRequestError)) {
+      return;
+    }
+    if (currentCallRequestResult) {
+      connector.approveRequest({
+        id: currentCallRequestId,
+        result: currentCallRequestResult,
+      });
+    }
+    if (currentCallRequestError) {
+      connector.rejectRequest({
+        id: currentCallRequestId,
+        error: { message: currentCallRequestError.message },
+      });
+    }
+    setCurrentCallRequestId();
+    setCurrentCallRequestResult();
+    setCurrentCallRequestError();
+  }, [
+    currentCallRequestId,
+    currentCallRequestError,
+    currentCallRequestResult,
+    connector,
+  ]);
+
+  useEffect(() => {
+    if (initDone) {
+      return;
+    }
+
+    const init = async () => {
+      setSession((await getWCSession()) || {});
+      setInitDone(true);
+    };
+
+    init();
+  }, [initDone]);
+
+  const { account } = useSelector(
+    accountScreenSelector({
+      params: { accountId: session.accountId },
+    }),
+  );
+  useEffect(() => {
+    if (
+      account &&
+      session.session &&
+      status === STATUS.DISCONNECTED &&
+      navigationRef.current
+    ) {
+      connect({ account });
+
+      navigate(NavigatorName.Base, {
+        screen: ScreenName.WalletConnectConnect,
+        params: {
+          accountId: account.id,
+        },
+      });
+    }
+  });
+
+  useEffect(() => {
+    if (!initDone) {
+      return;
+    }
+    saveWCSession(session);
+  }, [session, initDone]);
+
+  return (
+    <context.Provider
+      value={{
+        connect,
+        disconnect,
+        status,
+        error,
+        currentCallRequestId,
+        setCurrentCallRequestResult,
+        setCurrentCallRequestError,
+        dappInfo,
+        approveSession: approveSession && approveSession.fn,
+        initDone,
+        hasSession: Object.keys(session).length > 0,
+        // eslint-disable-next-line no-underscore-dangle
+        socketReady: connector?._transport?._socket?.readyState === 1,
+      }}
+    >
+      {children}
+    </context.Provider>
+  );
+};
+
+export default Provider;
```

Comment:
I have the feeling that this WalletConnect/Provider should be moved to live-common.

except some specific part it's not specific to mobile. we could do something similar to what we did for the "BridgeSync" in future if we generalize with desktop. (e.g. https://github.com/LedgerHQ/ledger-live-mobile/blob/develop/src/bridge/BridgeSyncContext.js which connect the specific implementation of mobile to use the generic's provider implemented in live-common)

Reply:
Yes I think we can move it to ledger live common

Reply:
some things need to be put out though, navigation logic and account selector.